addons/web/static/src/js/views/basic/basic_renderer.js

```diff
@@ -816,6 +816,9 @@ var BasicRenderer = AbstractRenderer.extend(WidgetAdapterMixin, {
             self._registerModifiers(node, record, widget);
             widget.$el.addClass('o_widget');
             $el.replaceWith(widget.$el);
+            if (self._isInDom && widget.on_attach_callback) {
```

Comment:
I came across that one when having a form view with chatter.
shouldn't we call attachcallback after renderView as a whole instead ?

Reply:
Looks more intuitive to invoke `on_attach_callback()` close to when we render the new widget and renderer is in DOM. Also invocation of `renderView` doesn't necessarily mean the renderer is in DOM, in fact we had to not assume that in `mail/form_renderer` because that was not the case most of the time on loading chatter for the 1st time.